src/utils/user-list-helpers.js

```diff
@@ -0,0 +1,259 @@
+  }
+
+  newUserPodUrl = newUserPodUrl || webId.split('profile')[0];
+
+  const newUserThing = buildThing(createThing({ name: `${username}` }))
+    .addStringNoLocale(RDF_PREDICATES.Person, `${givenName} ${familyName}`)
+    .addStringNoLocale(RDF_PREDICATES.givenName, givenName)
+    .addStringNoLocale(RDF_PREDICATES.familyName, familyName)
+    .addUrl(RDF_PREDICATES.identifier, webId)
+    .addUrl(RDF_PREDICATES.URL, newUserPodUrl)
+    .build();
+
+  solidDataset = setThing(solidDataset, newUserThing);
+
+  await saveSolidDatasetAt(`${userContainerUrl}userlist.ttl`, solidDataset, {
+    fetch: session.fetch
+  });
+
+  const userList = await getUsersFromPod(session, podUrl);
+  return userList;
+};
+
+/**
+ * Function that fetches a user's last active time on their Solid Pod
+ *
+ * @memberof utils
+ * @function getUserListActivity
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @param {userListObject[]} userList - An array of {@link userListObject}
+ * which stores the name and their Pod URL
+ * @returns {Promise<userListObject[]>} Promise - An array of users with last active
+ * time included to user list
+ */
+
+export const getUserListActivity = async (session, userList) => {
+  const userListWithTime = await Promise.all(
+    userList.map(async (user) => {
+      const activityUrl = `${user.podUrl}public/active.ttl`;
+      try {
+        const solidDataset = await getSolidDataset(activityUrl, {
+          fetch: session.fetch
+        });
+        const activeThing = getThingAll(solidDataset)[0];
+        const updatedUser = user;
+        updatedUser.dateModified = getDatetime(activeThing, RDF_PREDICATES.dateModified);
+        return updatedUser;
+      } catch {
+        return user;
+      }
+    })
+  );
+
+  return userListWithTime;
+};
+
+/**
+ * Function that updates a user's last active time on Solid Pod
+ *
+ * @memberof utils
+ * @function updateUserActivity
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @returns {Promise} Promise - Updates last active time of user to lastActive.ttl
+ */
+
+export const updateUserActivity = async (session, podUrl) => {
+  const activityDocUrl = `${podUrl}public/active.ttl`;
+  let activityDataset;
+
+  const updateAndSaveActivity = async () => {
+    const activity = buildThing(createThing({ name: 'active' }))
+      .addDatetime(RDF_PREDICATES.dateModified, new Date())
+      .build();
+
+    activityDataset = setThing(activityDataset, activity);
+
+    await saveSolidDatasetAt(activityDocUrl, activityDataset, {
+      fetch: session.fetch
+    });
+  };
+
+  try {
+    activityDataset = await getSolidDataset(activityDocUrl, { fetch: session.fetch });
+    await updateAndSaveActivity();
+  } catch {
+    activityDataset = createSolidDataset();
+    await updateAndSaveActivity();
+    const accessObject = {
+      read: true,
+      append: true,
+      write: true,
+      control: true
+    };
+    const publicAccessObject = {
```

Comment:
`activity.ttl` should be readable by anyone

Reply:
Yeah, we should make active.ttl private in the future with a PASS specific container with read access for all PASS users. For now, we'll just leave that in public.